pkg/gce-pd-csi-driver/controller.go

```diff
@@ -391,22 +393,34 @@ func (gceCS *GCEControllerServer) CreateVolume(ctx context.Context, req *csi.Cre
 		}
 		disk, err = createSingleZoneDisk(ctx, gceCS.CloudProvider, name, zones, params, capacityRange, capBytes, snapshotID, volumeContentSourceVolumeID, multiWriter)
 		if err != nil {
-			return nil, LoggedError("CreateVolume failed to create single zonal disk "+name+": ", err)
+			// Emit metric for expected disk type from storage class
+			if params.DiskType != "" {
```

Comment:
Can we use a defer statement to log the metrics, so that we don't miss any new return branches we could add inthe future?

Reply:
So something like this?
```
	if params.DiskType != "" {
		defer gceCS.Metrics.RecordOperationErrorMetrics("CreateVolume", err, params.DiskType)
	}
```
Could you please share an example of "so that we don't miss any new return branches we could add" for me to understand better.